zipline/assets/_assets.pyx

```diff
@@ -299,6 +314,75 @@ cdef class Future(Asset):
         super_dict['contract_multiplier'] = self.contract_multiplier
         return super_dict
 
+"""
+TODO MB - rework this we need a Currency object and we need a FX pair
+
+ok we should use this a CurrencyPair, Currency could just be an enum?
+"""
+
+cdef class CurrencyPair(Asset):
+    cdef readonly object pair # 'USDGBP' for example
+    cdef readonly object base # 'USD'
+    cdef readonly object quote # 'GBP'
+    cdef readonly int multiplier # Factor to multiply 10^cvf to get the actual
+    # amount traded so cvf=4 would mean multiply 10^4 = 10,000
+
+    def __cinit__(self,
+                  int sid,  # sid is required
+                  object symbol="",
+                  object pair="",
+                  object base="",
+                  object quote="",
+                  object start_date=None,
+                  int multiplier=1
+                  ):
+
+        self.pair = pair
+        self.base = base
+        self.quote = quote
+        self.start_date = start_date
+        self.multiplier = multiplier
+
+    def __str__(self):
+        if self.symbol:
+            return 'Currency(%d [%s])' % (self.sid, self.symbol)
+        else:
+            return 'Currency(%d)' % self.sid
+
+    def __repr__(self):
+        attrs = ('symbol', 'pair', 'base', 'quote',
+                 'start_date', 'multiplier')
+        tuples = ((attr, repr(getattr(self, attr, None)))
+                  for attr in attrs)
+        strings = ('%s=%s' % (t[0], t[1]) for t in tuples)
+        params = ', '.join(strings)
+        return 'Currency(%d, %s)' % (self.sid, params)
```

Comment:
The __str__ and __repr__ results should be consistent with the class name
CurrencyPair vs Currency